src/crawlers/statistics.js

```diff
@@ -44,21 +56,50 @@ export default class Statistics {
         checkParamOrThrow(logIntervalSecs, 'options.logIntervalSecs', 'Number');
         checkParamOrThrow(logMessage, 'options.logMessage', 'String');
 
-
+        this.log = defaultLog.child({ prefix: 'Statistics' });
         this.logIntervalMillis = logIntervalSecs * 1000;
         this.logMessage = logMessage;
+        this.keyValueStore = null;
+        // assign an id while incrementing so it can be saved/restored from KV
+        this.id = Statistics.id++;
+        this.persistStateKey = `STATISTICS_${this.id}_STATE`;
+
+        // initialize by "resetting"
+        this.reset();
+    }
+
+    /**
+     * Set the current statistic instance to pristine values
+     */
+    reset() {
+        if (!this.jobRetryHistogram) {
+            this.jobRetryHistogram = [];
+        } else {
+            this.jobRetryHistogram.length = 0;
+        }
 
-        this.jobRetryHistogram = [];
         this.finishedJobs = 0;
         this.failedJobs = 0;
-        this.jobsInProgress = new Map();
+
+        if (!this.jobsInProgress) {
+            this.jobsInProgress = new Map();
+        } else {
+            this.jobsInProgress.clear();
```

Comment:
Same here, not sure if the performance improvement, if any, is worth the extra code.